File: Cinema-app/services/RoomService.cs
using CinemaApp.Models;
using Cinema_app.model;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Linq;

namespace Cinema_app.Services
{
    public class RoomService
    {
        private readonly CinemaDbContext _context;

        public RoomService(CinemaDbContext context)
        {
            _context = context;
        }

        // Add a new room
        public void AddRoom(Room room)
        {
            _context.Rooms.Add(room);
            _context.SaveChanges();
        }

        // Get all rooms
        public List<Room> GetAllRooms()
        {
            return _context.Rooms.Include(r => r.Cinema).ToList();
        }

        // Get room by ID
        public Room GetRoomById(int id)
        {
            return _context.Rooms.Include(r => r.Cinema).FirstOrDefault(r => r.Id == id);
        }

        // Update a room
        public void UpdateRoom(Room updatedRoom)
        {
            var existingRoom = _context.Rooms.FirstOrDefault(r => r.Id == updatedRoom.Id);
            if (existingRoom != null)
            {
                existingRoom.RoomNumber = updatedRoom.RoomNumber;
                existingRoom.Capacity = updatedRoom.Capacity;
                existingRoom.CinemaId = updatedRoom.CinemaId;
                existingRoom.Features = updatedRoom.Features;

                _context.SaveChanges();
            }
        }

        // Delete a room by ID
        public void DeleteRoom(int id)
        {
            var room = _context.Rooms.FirstOrDefault(r => r.Id == id);
            if (room != null)
            {
                _context.Rooms.Remove(room);
                _context.SaveChanges();
            }
        }

        // Get rooms by Cinema ID
        public List<Room> GetRoomsByCinemaId(int cinemaId)
        {
            return _context.Rooms.Where(r => r.CinemaId == cinemaId).ToList();
        }

        // Search rooms by features
        public List<Room> SearchRooms(string searchTerm)
        {
            return _context.Rooms
                           .Where(r => r.Features.Contains(searchTerm))
                           .ToList();
        }
    }
}

File: web-app/src/services/RoomService.js
import axios from 'axios';

const API_URL = 'https://localhost:7189/api/Rooms';  // Adjust to match your API

// Get all rooms
export const getAllRooms = async () => {
  try {
    const response = await axios.get(API_URL);
    return response.data;
  } catch (error) {
    throw error.response ? error.response.data : 'Error fetching rooms';
  }
};

// Get room by ID
export const getRoomById = async (id) => {
  try {
    const response = await axios.get(`${API_URL}/${id}`);
    return response.data;
  } catch (error) {
    throw error.response ? error.response.data : 'Error fetching room';
  }
};

// Add a new room
export const addRoom = async (room) => {
  try {
    const response = await axios.post(API_URL, room);
    return response.data;
  } catch (error) {
    throw error.response ? error.response.data : 'Error adding room';
  }
};

// Update room by ID
export const updateRoomById = async (id, room) => {
  try {
    const response = await axios.put(`${API_URL}/${id}`, room);
    return response.data;
  } catch (error) {
    throw error.response ? error.response.data : 'Error updating room';
  }
};

// Delete room by ID
export const deleteRoomById = async (id) => {
  try {
    const response = await axios.delete(`${API_URL}/${id}`);
    return response.data;
  } catch (error) {
    throw error.response ? error.response.data : 'Error deleting room';
  }
};

// Get rooms by cinema ID
export const getRoomsByCinemaId = async (cinemaId) => {
  try {
    const response = await axios.get(`${API_URL}/byCinema/${cinemaId}`);
    return response.data;
  } catch (error) {
    throw error.response ? error.response.data : 'Error fetching rooms by cinema';
  }
};

// Search rooms by features
export const searchRooms = async (searchTerm) => {
  try {
    const response = await axios.get(`${API_URL}/search`, { params: { searchTerm } });
    return response.data;
  } catch (error) {
    throw error.response ? error.response.data : 'Error searching rooms';
  }
};
